tcms/testcases/models.py

```diff
@@ -391,11 +393,11 @@ def latest_text(self, text_required=True):
         if not text_required:
             text = text.defer('action', 'effect', 'setup', 'breakdown')
         qs = text.order_by('-case_text_version')[0:1]
-        return NoneText if len(qs) == 0 else qs[0]
+        return NoneText if not qs else qs[0]
```

Comment:
Revert this change. Here `qs` is an iterator which is never None. In this context `len()` will force the queryset to be evaluated and return the number of objects. I'm not confident enough that this change will always work as intended to.

In addition the expression `filter()[0:1]` is completely unnecessary, From what I can tell it takes the first element (after ordering) which is done by `.first()` as well. This entire method can be refactored in a more appropriate way.

Additionally I think some of these `latest_text` methods can be removed so need to inspect the a bit closer.